src/Entity/Package.php

```diff
@@ -687,6 +687,23 @@ public function getType(): string|null
         return $this->type;
     }
 
+    public function getInstallCommand(): string
+    {
+        $command = 'create-project';
+
+        if ('project' !== $this->getType()) {
+            $command = 'require';
+
+            /** @var Version $version */
+            $version = $this->getVersions()->first();
```

Comment:
not sure which version to use ...

Reply:
The `view_package` template already receive a `version` variable (which can be `null`). this is the version that should be used for the install command IMO (which would be done by keeping the command in the template as it is done right now, but adding a `{% if version and version.hasDevTag %} --dev{% endif %}`)

Reply:
yea, i know that moving it from the template to PHP feels like the more complex solution at first, but i tend to kill every if in twig that i can replace with sth that is easily unit tested.

but sure, that's a debatable design decision

Reply:
Twig already has the right version to use passed to it by the controller. The entity does not have it. So moving this to the entity itself looks like the bad decision IMO.

Reply:
wasn't aware of that controller code to sort out the most relevant version - thanks for that pointer!
will give it another thought later :+1:

Reply:
IMO this function should just accept a version as argument, and then the template can pass it in.